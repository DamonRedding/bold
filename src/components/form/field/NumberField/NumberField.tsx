import React from 'react'

import { BaseFieldProps } from '../../finalForm/Field'
import { TextInputProps } from '../../input/TextInput/TextInput'
import { TextField } from '../TextField/TextField'

export const parse = value => value.replace(/[^\d]/g, '')

export interface NumberFieldProps extends BaseFieldProps<TextInputProps> {}

export class NumberField extends React.Component<NumberFieldProps> {
  render() {
    return <TextField parse={parse} {...this.props} />
  }
}
