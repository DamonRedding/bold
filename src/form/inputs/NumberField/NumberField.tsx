import React from 'react'

import { TextInputProps } from '../../../components/TextInput'
import { BaseFieldProps } from '../../Field'
import { TextField } from '../TextField/TextField'

export const parse = value => value.replace(/[^\d]/g, '')

export interface NumberFieldProps extends BaseFieldProps<TextInputProps> {}

export class NumberField extends React.Component<NumberFieldProps> {
  render() {
    return <TextField parse={parse} {...this.props} />
  }
}
