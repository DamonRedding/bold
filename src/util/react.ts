import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
