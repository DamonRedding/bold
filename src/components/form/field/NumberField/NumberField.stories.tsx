import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { withForm } from '../../../../stories-addons/withForm'

import { NumberField } from './NumberField'

storiesOf('Form/Fields', module)
    .addDecorator(withForm())
    .add('NumberField', () =>
        <NumberField name='quantidade' label='Quantidade' placeholder='Quantidade' />
    )
