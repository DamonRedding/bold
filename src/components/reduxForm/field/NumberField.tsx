import * as React from 'react'
import { TextField } from './TextField/TextField'

export const normalize = value => value.replace(/[^\d]/g, '')

export interface NumberFieldProps {
    name: string
}

export class NumberField extends React.Component<NumberFieldProps> {

    render() {
        return (
            <TextField {...this.props} normalize={normalize} />
        )
    }

}
