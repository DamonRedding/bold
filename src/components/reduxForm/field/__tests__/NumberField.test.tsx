import * as React from 'react'
import { mount } from 'enzyme'
import { normalize, NumberField } from '../NumberField'
import { shallowRenderAndMatch, testForm } from '../../../../__tests__/test.defaults'

describe('NumberField', () => {
    describe('normalize', () => {
        it('só deve aceitar números', () => {
            expect(normalize('abcdefgh999!@.a')).toEqual('999')
        })
    })

    describe('NumberField', () => {
        it('render', () => {
            shallowRenderAndMatch(testForm(() => <NumberField name='number' />))
        })

        it('deve aceitar somente números', () => {
            const wrapper = mount(testForm(() => <NumberField name='number' />))
            const inputWrapper = wrapper.find('[name="number"]')
            inputWrapper.simulate('change', { target: { value: 'abc123a!' } })
            expect(inputWrapper.props().value).toEqual('123')
        })
    })
})
