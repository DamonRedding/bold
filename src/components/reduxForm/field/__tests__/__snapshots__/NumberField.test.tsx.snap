// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberField NumberField render 1`] = `<ReduxForm />`;
